src/JWT/JwtValidator.cs

```diff
@@ -87,6 +88,24 @@ public void Validate(string decodedPayload, IAsymmetricAlgorithm alg, byte[] byt
                 throw ex;
         }
 
+        public void Validate(byte[][] keys, string decodedPayload, ISymmetricAlgorithm alg, byte[] bytesToSign, byte[] decodedSignature)
+        {
+            if (alg.Key == null && keys is null)
+                throw new ArgumentNullException(nameof(keys));
+            if (alg.Key == null && !AllKeysHaveValues(keys))
+                throw new ArgumentOutOfRangeException(nameof(keys));
+
+            // the signature on the token, with the leading =
+            var rawSignature = Convert.ToBase64String(decodedSignature);
+
+            // the signatures re-created by the algorithm, with the leading =
+            var recreatedSignatures = keys != null
+                ? keys.Select(key => Convert.ToBase64String(alg.Sign(key, bytesToSign))).ToArray()
+                : new string[] { Convert.ToBase64String(alg.Sign(null, bytesToSign)) };

```

Comment:
a nice occasion to use "array literals", e.g. `[Convert.ToBase64String(alg.Sign(null, bytesToSign))]`